client/modules/IDE/components/Header/MobileNav.jsx

```diff
@@ -0,0 +1,388 @@
+            />
+          </div>
+        )}
+      </Options>
+    </Nav>
+  );
+};
+
+const AccoutnMenu = () => {
+  const user = useSelector((state) => state.user);
+  const dispatch = useDispatch();
+
+  // use the useNavigate hook in the react router v6
+  const history = useHistory();
+  const navigate = (url) => history.push(url);
+
+  return (
+    <div>
+      <IconButton icon={AccountIcon} />
+      <ul>
+        <li className="user">{user.username}</li>
+        <li>
+          <button
+            onClick={() => {
+              navigate(`/${user.username}/sketches`);
+            }}
+          >
+            My Stuff
+          </button>
+        </li>
+        <li>
+          <button
+            onClick={() => {
+              navigate(`/account`);
+            }}
+          >
+            Settings
+          </button>
+        </li>
+        <li>
+          <button onClick={() => dispatch(logoutUser())}>Log Out</button>
+        </li>
+      </ul>
+    </div>
+  );
+};
+
+const MoreMenu = () => {
+  // TODO: use selectRootFile selector
+  const rootFile = useSelector(
+    (state) => state.files.filter((file) => file.name === 'root')[0]
+  );
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { newSketch, saveSketch } = useSketchActions();
+  const history = useHistory();
+  const navigate = (url) => history.push(url);
+
+  const cmRef = useContext(CmControllerContext);
+
+  return (
+    <div>
+      <IconButton icon={MoreIcon} />
+      <ul>
+        <b>{t('Nav.File.Title')}</b>
+        <li>
+          <button onClick={newSketch}>{t('Nav.File.New')}</button>
+        </li>
+        <li>
+          <button onClick={() => saveSketch(cmRef.current)}>
+            {t('Common.Save')}
+          </button>
+        </li>
+        <li>
+          <button
+            onClick={() => {
+              navigate(`/p5/sketches`);
+            }}
+          >
+            {t('Nav.File.Examples')}
+          </button>
+        </li>
+        <b>{t('Nav.Edit.Title')}</b>
+        <li>
+          <button onClick={cmRef.current?.tidyCode}>
+            {t('Nav.Edit.TidyCode')}
+          </button>
+        </li>
+        <li>
+          <button onClick={cmRef.current?.showFind}>
+            {t('Nav.Edit.Find')}
+          </button>
+        </li>
+        <b>{t('Nav.Sketch.Title')}</b>
+        <li>
+          <button onClick={() => dispatch(newFile(rootFile.id))}>
+            {t('Nav.Sketch.AddFile')}
+          </button>
+        </li>
+        <li>
+          <button onClick={() => dispatch(newFolder(rootFile.id))}>
+            {t('Nav.Sketch.AddFolder')}
+          </button>
+        </li>
+        {/* TODO: Add Translations */}
+        <b>Settings</b>
+        <li>
+          <button
+            onClick={() => {
+              dispatch(openPreferences());
+            }}
+          >
+            Preferences
+          </button>
+        </li>
+        <li>
+          <button>Language</button>
+        </li>
```

Comment:
What are you picturing for the language selection?  Probably a question for @shujuuu.  We could open a modal but it might be cool if we can use a native `<select>` inline with the menu so that clicking on the language brings up the phone’s standard UI for selecting for dropdown menus.

Reply:
yeah I'll ask her, will update this in future

Reply:
Probably the quickest/easiest way to get something that's usable would be to open a modal.  No Redux state, use a `languageModalOpen` component state in `MoreMenu` and conditionally render the modal if it's true.  Not sure about the content of the modal but maybe just a native `<select>` for right now even if it's ugly.  This wouldn't be the final product but it would allow us to go live with the `MobileNav` without creating any usability issues.  We just need to make sure that there is some way to select a language.

IMO this and the "Find" search box are the only things that we need to address right now; and everything else is just a "nice to have" which we can revisit later.